tests/Unit/Auth/CanRootDirectiveTest.php

```diff
@@ -0,0 +1,180 @@
+<?php declare(strict_types=1);
+
+namespace Auth;
+
+use Illuminate\Contracts\Auth\Access\Gate;
+use Tests\Unit\Auth\CanDirectiveTestBase;
+use Tests\Utils\Models\User;
+use Tests\Utils\Policies\UserPolicy;
+
+class CanRootDirectiveTest extends CanDirectiveTestBase
+{
+    public static function getSchema(string $commonArgs): string
+    {
+        return /** @lang GraphQL */ "
+            type Query {
+                user: User!
+                    @mock
+            }
+
+            type User {
+                name(foo: String): String @canRoot({$commonArgs})
+            }
+        ";
+    }
+
+    protected function getQuery(): string
+    {
+        return /** @lang GraphQL */ '
+            query ($foo: String) {
+                user {
+                    name (foo: $foo)
+                }
+            }
+        ';
+    }
```

Comment:
repeats parent?

Reply:
No. 
In CanRoot we apply argument to field `name`, not to query `user`. Otherwise test with `injectArgs: true` won't work.